### PR TITLE
Add search history to member and streamfile searching

### PR DESCRIPTION
**Pull request by chrjorgensen**
### Changes

This PR will add a history of search terms, which can be selected from a list when entering a search term in member and streamfiles searches.

Screenshot from 2023-09-09 21-14-24

Screenshot from 2023-09-09 21-14-52

The input dialog has been changed into a quickpick list, where the user can enter a new search term or select one of the previous search terms. The previous search terms are stored in global storage, thus common to all connections.

@sebjulliand Three new messages have been added - can you translate them into French, please?

```
  'ifsBrowser.searchIFS.previousSearches': `Previous search terms`,
  'ifsBrowser.searchIFS.placeholder': `Enter search term or select one of the previous search terms.`,
  'ifsBrowser.searchIFS.placeholder2': `Enter search term.`,
```

### Checklist

* [x] have tested my change
* [x] eslint is not complaining

### CODE REVIEW

File: src/views/ifsBrowser.js [hunk 1]
```diff
@@ -687,6 +705,37 @@ module.exports = class IFSBrowser {
...
+
+        let results = await Search.searchIFS(getInstance(), searchPath, searchTerm);
+
+        if (results.length > 0) {
+          results = results.map(a => ({ ...a, label: path.posix.relative(searchPath, a.path) }));
+          setSearchResults(searchTerm, results.sort((a, b) => a.path.localeCompare(b.path)));
+
+        } else {
+          vscode.window.showInformationMessage(t(`ifsBrowser.doSearchInStreamfiles.noResults`, searchTerm, searchPath));
+        }
```
Comment (sebjulliand) [inline src/views/ifsBrowser.js:732]: As `searchIFS` always returns an array, you can safely perform the `map` and `sort` operations here all at once and make `result` a `const`.
```suggestion
        const results = (await Search.searchIFS(getInstance(), searchPath, searchTerm))
          .map(a => ({ ...a, label: path.posix.relative(searchPath, a.path) }))
          .sort((a, b) => a.path.localeCompare(b.path));

        if (results.length > 0) {
          results = results;
          setSearchResults(searchTerm, results);
        } else {
          vscode.window.showInformationMessage(t(`ifsBrowser.doSearchInStreamfiles.noResults`, searchTerm, searchPath));
        }
```

**Review by sebjulliand — CHANGES_REQUESTED** [review]
Nice feature @chrjorgensen !
I added (and fixed) the FR translations. I also made a few styling suggestions 😉
Check it out and once you're done we can safely merge this.

File: src/views/ifsBrowser.js [hunk 2]
```diff
@@ -538,35 +539,52 @@ module.exports = class IFSBrowser {
...
-                  vscode.window.showInformationMessage(t(`ifsBrowser.searchIFS.noResults`, searchTerm, searchPath));
-                }
-              });
-
-            } catch (e) {
-              vscode.window.showErrorMessage(t(`ifsBrowser.searchIFS.errorMessage`));
+          // let searchTerm = await vscode.window.showInputBox({
+          //   prompt: t(`ifsBrowser.searchIFS.prompt2`, searchPath)
+          // });
+          let list = [...GlobalStorage.get().getPreviousSearchTerms() || []];
```
[the remark this answers is not in the document]
Comment (chrjorgensen) [reply src/views/ifsBrowser.js]: Almost done - the list array is changed later, e.g. when clearing the list, so can't be a `const`...

**Comment by chrjorgensen** [comment]
@sebjulliand Followed your advices - almost... 😉

File: src/views/ifsBrowser.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (sebjulliand) [reply src/views/ifsBrowser.js]: Indeed, fair enough 😄

**Review by sebjulliand — APPROVED** [review]
Works fine! Let's merge!

**Pull request merged**